core/commands/add.go

```diff
@@ -277,16 +318,24 @@ func addDir(n *core.IpfsNode, dir files.File, out chan interface{}, progress boo
 			break
 		}
 
-		node, err := addFile(n, file, out, progress, false)
-		if err != nil {
+		node, err := addFile(n, file, out, progress, false, hidden, localIgnorePatterns)
+		if _, ok := err.(*hiddenFileError); ok {
+			// hidden file error, set the node to nil for below
+			node = nil
+		} else if _, ok := err.(*ignoreFileError); ok {
+			// ignore file error, set the node to nil for below
+			node = nil
+		} else if err != nil {
```

Comment:
Tested this one on my box and it was working correctly. The recursive version eats the errors, but adding a specific file returns the error.

```bash
 > echo "ignore*" > .ipfsignore
 > ipfs add ignore_match.txt
Error: ignore_match.txt is an ignored file
 > ipfs add . -r
... list of stuff not including ignore_match.txt
```